### PR TITLE
ipn/ipnlocal: make tests pass when offline

### PR DESCRIPTION
**Pull request by maisem**
Updates #cleanup

### CODE REVIEW

File: ipn/ipnlocal/local.go [hunk 1]
```diff
@@ -91,6 +93,14 @@ import (
 	"tailscale.com/wgengine/wgcfg/nmcfg"
 )
 
+var lazyInTest lazy.SyncValue[bool]
+
+func inTest() bool {
+	return lazyInTest.Get(func() bool {
+		return flag.Lookup("test.v") != nil
+	})
+}
```
Comment (bradfitz) [inline ipn/ipnlocal/local.go:102]: sure, but this should live elsewhere like we have in corp. Then corp can use the oss copy too.

**Review by bradfitz — APPROVED** [review]

File: ipn/ipnlocal/local.go (hunk 1 above)
Comment (maisem) [reply ipn/ipnlocal/local.go:102]: yeah, we have a few copies of this in oss. Any package preferences?

Comment (bradfitz) [reply ipn/ipnlocal/local.go:102]: We have `util/cibuild` for just `func On() bool`.

Something beside that? `util/testenv`? `testenv.InTest()` ?

Comment (maisem) [reply ipn/ipnlocal/local.go:102]: Sent #8815

**Pull request merged**